### PR TITLE
refactor of state tracking for loop mounting

### PR DESCRIPTION
**Pull request by jaromil**
testing a small refactor

simplified function calls for tracking of loop mount by using global
variables whose scope is limited to execution, most computation is now
included in the `is_valid_tomb` function.

### CODE REVIEW

**Comment by Narrat** [comment]
Didn't look at the changes, but it kinda looks like a permission problem is still there
If there wasn't any loop device initialized before, it will try to do such a thing and fail because of privileges.

```
$ tomb open test -k lalala.key                                                         
(...)
losetup: unused loop-device couldn't be found: unsufficient permissions
basename: missing Operand
„basename --help“ for more Information

[sudo] Enter password for Superuser-Rights
```
and on close:
```
$ tomb close                                                                                             
tomb  .  Tomb [test], mounted at /run/media/omnius/test, is getting closed
losetup: /dev/: use of the device failed: no such device found
tomb (*) Tomb [test] closed: (...)
```
And it didn't get closed completly. Like the reported issue with Nautilus: The mapper or loop still remained.
Curiously next time everything worked fine. I suppose because of the existing loop device.

File: tomb [hunk 1]
```diff
@@ -597,67 +569,60 @@ is_valid_tomb() {
...
-render_mapper() {
-	[[ "$tombpath" == "" ]] &&
-		_failure "cannot render mapper: missing \$tombpath"
-	[[ "$TOMBNAME" == "" ]] &&
-		_failure "cannot render mapper: missing \$TOMBNAME"
-	local maphash=`realpath $tombpath | sha256sum -z`
-	mapper="tomb.$TOMBNAME.${maphash[(w)1]}.loop"
+	# checks if Tomb already mounted (or we cannot alter it)
+	local maphash=`realpath $TOMBPATH | sha256sum -z`
+	local nextloop=`losetup -f`
```
Comment (shaform) [inline tomb:602]: Right, it seems that if the loop device is not already present in `/dev/` then this line would cause a permission error.

Comment (jaromil) [reply tomb:602]: oddly enough tests have passed without problems

Comment (Narrat) [reply tomb:602]: But there the "error" also appears
```
(...)
tomb  .  Testing creation: lock

tomb [D] Identified caller: omnius (1000:100)
tomb [D] Tomb command: lock /tmp/test.tomb
tomb [D] Caller: uid[1000], gid[100], tty[/dev/pts/3].
tomb [D] Temporary directory: /tmp/zsh
tomb [D] is_valid_tomb /tmp/test.tomb
tomb [D] tomb file is readable
tomb [D] tomb file is a regular file
tomb [D] tomb file is not empty
tomb [W] File is not yet a tomb: /tmp/test.tomb
losetup: cannot find an unused loop device: Permission denied
basename: missing operand
Try 'basename --help' for more information.
tomb [D] Mapper: tomb.test.10abda9a4ec9d0e665fe25155685f2e9ddf8f98034aa1b35a8b3748099751b24.
tomb [D] tomb file is not currently in use
tomb  .  Valid tomb file found: /tmp/test.tomb
tomb  .  Commanded to lock tomb test.tomb
tomb [D] Tomb found: /tmp/test.tomb
tomb [D] Loop mounted on /dev/loop0
tomb  .  Checking if the tomb is empty (we never step on somebody else's bones).
tomb  .  Fine, this tomb seems empty.
tomb [D] load_key argument: /tmp/test.tomb.key
tomb [D] load_key: /tmp/test.tomb.key
tomb [D] is_valid_key
(...)
```
In the end it passed the test, but didn't work correctly

**Pull request merged**